### ISSUE TITLE
WP components audit: ClipboardButton & FormFileUpload

### ISSUE THREAD

**Issue by davewhitley** [WordPress/gutenberg#16543]
**Is your feature request related to a problem? Please describe.**

This issue is part of a review on the naming, structure, and composition of the UI components as brought up in https://github.com/WordPress/gutenberg/issues/16367

# [ClipboardButton](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/button-group)

![Screen Shot 2019-07-09 at 3 15 33 PM](https://user-images.githubusercontent.com/618551/60919870-71b6b100-a25c-11e9-8e4b-2ad7aa8d8bdd.png)

## Audit 
Opportunities I came across while reviewing this component.

- I expected this component to be combined with the Button component.

## Grouping
- I expect this component to live in a Buttons category, instead of a top level item.

## Suggestions

- Consider renaming to `CopyButton` because this more directly relates to the component and it's simpler language. The button is used to _copy_ something.
- Consider de-emphasizing this component and not list it as a base UI component.

# [FormFileUpload](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/toolbar)

![Screen Shot 2019-07-09 at 3 25 09 PM](https://user-images.githubusercontent.com/618551/60920418-c3ac0680-a25d-11e9-99e8-93d0e97f05cb.png)

## Audit 
Opportunities I came across while reviewing this component.

- The name was a bit obtuse for me to understand quickly at first. Is this a button?
- This component seems to render an IconButton with an additional text label aligned to the right. It looks like a regular Button. This also makes me question whether or not I can use Button props like isPrimary on this component.
- I question whether this component should be a separate component or consolidated into Button.

## Grouping

- I expect this component to live in a Buttons category, instead of a top level item.

## Suggestions

- Consider renaming to `UploadButton` because this more directly relates to the component and it's simpler language.
- If possible, consolidate this button with the Button component. I noticed in other components libraries, they don't have a separate upload button component.
- If we need to keep it as a separate component, consider de-emphasizing this component and not list it as a base UI component.

# Feedback

The feedback I'm looking for is related to naming, structure, and composition. I'm not looking for visual feedback of the components. I'm also not examining each prop for each component; only the ones that I think are relevant for the audit.

- What do you think about combining these components into Button?
- If the component is not adding much functionality, we can show an example of how to compose a Button component with upload functionality in the developer docs, for example.

**Comment by drw158** [comment]
The [Evergreen](https://evergreen.segment.com/components/button) component library has a BackButton component, which is considered a "**preset** based on the Button component."

Can ClipboardButton and FormFileUpload be considered the same and are they implemented that way in code?

**Comment by paaljoachim** [comment]
Hey Dave @davewhitley 

Can you clarify how design can help out with this issue?
Thanks.

**Comment by davewhitley** [comment]
Not visual design, but design in a system sense. More background here https://make.wordpress.org/design/2019/08/12/project-reviewing-wordpress-components/

**Comment by paaljoachim** [comment]
Ahh ok. Thanks for the link!

@melchoyce @mapk

**Comment by mapk** [comment]
I'm closing out the Component Audit issues for now.
